### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by fiunchinho**
Adds e2e tests to make sure aws-load-balacer-controller app behaves as expected.

### CODE REVIEW

**Comment by paurosello** [comment]
seems good! not sure the aws-load-balancer-controller adds some finalizer to the Service but if it does maybe we could add a test to check for it just to make sure it's the one provisioning the LB but already 100% better than current status

**Comment by fiunchinho** [comment]
Pipeline is currently failing with
```
failed to create AWSMachine instance: failed to find ami: found no AMIs with the name: \"flatcar-stable-3975.2.0-kube-v1.29.8-gs\""
```

**Comment by fiunchinho** [comment]
Blocked until [this is](https://github.com/giantswarm/apptest-framework/pull/173) released.

File: tests/e2e/suites/basic/basic_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+				}).
+					WithTimeout(6 * time.Minute).
+					WithPolling(5 * time.Second).
+					Should(BeTrue())
+
+				By("adding aws-load-balancer-controller finalizer to service")
+
+				// We make sure the `Service` has the finalizer added by the aws-load-balancer-controller
+				Eventually(func() (bool, error) {
+					return serviceHasFinalizer(nginxIngressApp.GetInstallNamespace(), "ingress-nginx-controller", "service.k8s.aws/resources")
```
Comment (fiunchinho) [inline tests/e2e/suites/basic/basic_suite_test.go:91]: I'm checking the finalizer to be sure that it's the controller the one creating the LB.

File: tests/e2e/go.mod [hunk 2]
```diff
@@ -0,0 +1,191 @@
+module e2e
+
+go 1.23.1
+
+replace github.com/imdario/mergo => github.com/imdario/mergo v0.3.16
+
+replace github.com/alessio/shellescape => al.essio.dev/pkg/shellescape v1.4.2
```
Comment (AndiDog) [inline tests/e2e/go.mod:7]: [minor]

Can we comment why? Else it's hard to remove the `replace` directives.

**Review by AndiDog — APPROVED** [review]

File: tests/e2e/go.mod (hunk 2 above)
Comment (fiunchinho) [reply tests/e2e/go.mod:7]: [It's generated by `devctl`](https://github.com/giantswarm/devctl/blob/main/pkg/gen/input/apptest/internal/file/go.mod.template), I'm not sure why the replace is required.

Comment (fiunchinho) [reply tests/e2e/go.mod]: Opened https://github.com/giantswarm/devctl/pull/1103

**Review by AndiDog — APPROVED** [review]

**Pull request merged**